cli/src/semdep/parse_lockfile.py

```diff
@@ -123,6 +123,22 @@ def parse_lockfile_path(
 
     Also returns Optional DependencyParseError as second return value if there was a problem
     parsing the lockfile
+
+    Raises SemgrepError if the lockfile is not supported
+    """
+    file_changed_timestamp = os.stat(lockfile_path).st_mtime
+    return _parse_lockfile_path_helper(lockfile_path, file_changed_timestamp)
+
+
+@lru_cache(maxsize=1000)
+def _parse_lockfile_path_helper(
+    lockfile_path: Path, file_changed_timestamp: float
```

Comment:
maybe a dumb question: does file_changed_timestamp get used anywhere? or is it just that `@lru_cache` sees that the param changes and doesn't return the cached value?

Reply:
not used anywhere. the cache key are all arguments to a function + function name